src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraAcl.scala

```diff
@@ -0,0 +1,26 @@
+
+package org.broadinstitute.dsde.agora.server.business
+
+import org.broadinstitute.dsde.agora.server.business.AgoraPermissions._
+
+object AgoraPermissions {
+  val Read = 1 << 0
+  val Write = 1 << 1
+  val Create = 1 << 2
+  val Redact = 1 << 3
+  val Manage = 1 << 4
+  val ReadWrite = Read | Write
+  val All = Read | Write | Create | Redact | Manage
+}
+
+case class AgoraAcl(permissions: Int) {
+  def canRead: Boolean = (permissions & Read) != 0
+
+  def canWrite: Boolean = (permissions & Write) != 0
+
+  def canCreate: Boolean = (permissions & Create) != 0
+
+  def canRedact: Boolean = (permissions & Redact) != 0
+
+  def canManage: Boolean = (permissions & Manage) != 0
+}
```

Comment:
Ah ok I misunderstood what you were saying. It is fair point that if we aren't storing things we don't need to worry about the efficiency of a bit field. When I think permissions I think bitmask so that was my initial thought for implementation since it is probably the most common and mimics posix permissions.

Personally I don't find acl = AgoraAcl(Read | Write | Create |Redact) to be anymore onerous than Set(Read, Write, Create, Redact) in fact you could create a var arg constructor that can do just that AgoraAcl(Read, Write, Create, Redact)

For 2 I'm not sure it is much more complicated either just and the not value
so AgoraAcl(acl.permisions & ~Create & ~Redact).

If we ever did need to store them serialization becomes easier and you also get more type safety using a case class instead of a generic Set of ints.

Does anyone else have an opinion?